cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java

```diff
@@ -298,11 +299,27 @@ private List<Runnable> processNotification(ProgramHeartbeatTable programHeartbea
       return result;
     }
 
+    //At this time, it is expected that provisioning service should be up and running.
+    waitForProvisioningService(provisioningService);
```

Comment:
We just need to make sure that provisioningService is in RUNNING state when this service is starting. Why not calling this method here and not during startup time?